wikimedia/wikipedia-ios
# Bug fix - Native Editor inner formatting action

**Phabricator:** 
[N/A](https://phabricator.wikimedia.org/T347925)

### Notes
This is a bug I noticed when comparing inner formatting with Android and Code Mirror. When selecting formatted inner text (i.e. not the entire string), it it expected that it removes formatting by surrounding the section in opposing strings, e.g.:

From:
```Testing <s>Strikethrough text here</s>```

To:
```Testing <s>Strikethrough </s>text<s> here</s>```

If you were to highlight "text" and toggle off strikethrough.

This PR applies this logic for our current formatters, bold, italics and templates.

### Test Steps
1. In Staging on editor, select 3 words and bold them.
2. Highlight the second word and deselect Bold toolbar button
3. Confirm second word is surrounded by wikitext bold marks and is no longer formatted as bold. Select Bold toolbar button again.
4. Confirm second word is no longer surrounded by wikitext bold marks and is formatted as bold again.

### Screenshots/Videos

Before:
![out1](https://github.com/wikimedia/wikipedia-ios/assets/3620196/e0e0191e-d45a-4144-b937-5ecf04bac5ce)

After:
![out2](https://github.com/wikimedia/wikipedia-ios/assets/3620196/1065fce9-41fd-4d90-b20d-d03b4aec3999)
